mlflow/models/evaluation/base.py

```diff
@@ -775,81 +775,157 @@ def _start_run_or_reuse_active_run():
         yield active_run.info.run_id
 
 
-def _normalize_evaluators_and_evaluator_config_args(
-    evaluators,
-    evaluator_config,
-):
+def _resolve_default_evaluator(model_type, evaluator_config):
+    """
+    Determine which built-in evaluators should be used for the given model type by default.
+
+    Previously, MLflow evaluate API only had a single "default" evaluator used for all models like
+    classifier, regressor, etc. We split it into multiple built-in evaluators for different model
+    types for maintainability, but in order to maintain backward compatibility, we need to map
+    the "default" provided by users to the correct built-in evaluators.
+
+    Args:
+        model_type: A string describing the model type (e.g., "regressor", "classifier", …).
+        evaluator_config: A dictionary of additional configurations for the evaluator.
+    """
     from mlflow.models.evaluation.evaluator_registry import _model_evaluation_registry
 
+    builtin_evaluators = []
+    for evaluator_name in list(_model_evaluation_registry._registry.keys()):
+        if (
+            evaluator_name != "default"
+            and _model_evaluation_registry.is_builtin(evaluator_name)
+            and (evaluator := _model_evaluation_registry.get_evaluator(evaluator_name))
+            and evaluator.can_evaluate(
+                model_type=model_type, evaluator_config=evaluator_config or {}
+            )
+        ):
+            builtin_evaluators.append(evaluator_name)
+
+    # We should use DefaultEvaluator only if there is no other built-in evaluator applicable.
+    return builtin_evaluators or ["default"]
+
+
+# NB: We often pass around evaluator name, config, and its instance together. Ideally, the
+# evaluator class should have name and config as class attributes, however, it was not
+# designed that way. Adding them while keeping backward compatibility is not trivial.
+# So, we use a dataclass to bundle them together.
+@dataclass
+class EvaluatorBundle:
+    name: str
+    evaluator: ModelEvaluator
+    config: Dict[str, Any]
+
+
+def resolve_evaluators_and_configs(
+    evaluators: Union[str, List[str], None],
+    evaluator_config: Union[Dict[str, Any], None],
```

Comment:
A random idea. Unrelated this PR. Can discuss offline. What if `evaluators` take a list of dicts (= pair of name and config)?

```python
mlflow.evaluate(
    evaluators=[
        {"name": "...", coning: {...}}
    ],
    ...
)
```

Reply:
Yea I agree passing evaluators and configs separately makes the handling complex and confusing. Indeed there is an ongoing design discussion and we will probably pass evaluators instances (which includes config) not name, I think it will resolve the issue. But I think it's good idea to pass dict if we need to keep names as inputs🙂

Reply:
Passing evaluators instances also makes sense.